src/webbleabstraction.ts

```diff
@@ -99,11 +99,14 @@ export class WebBLEDevice extends EventEmitter implements IBLEAbstraction {
             return callback(buf);
         };
         this._characteristics[uuid].addEventListener("characteristicvaluechanged", this._listeners[uuid]);
-        for (const data of this._mailbox) {
+
+        const mailbox = Array.from(this._mailbox);
+        this._mailbox = [];
```

Comment:
copy mailbox content and reset mailbox. It prevent message being dropped if callback is "posting" to the mailbox again.